lms/product/moodle/_plugin/misc.py

```diff
@@ -1,11 +1,48 @@
-from lms.product.plugin.misc import MiscPlugin
+from lms.models import Assignment
+from lms.product.plugin.misc import AssignmentConfig, MiscPlugin
 
 
 class MoodleMiscPlugin(MiscPlugin):
     deep_linking_prompt_for_title = False
     # Moodle's deep linking flow is included in the activity creating one.
     # Removing our title prompt to avoid confusion with the one in the LMS.
 
+    def get_assignment_configuration(
+        self,
+        request,
+        assignment: Assignment | None,
+        _historical_assignment: Assignment | None,
+    ) -> AssignmentConfig:
+        deep_linked_config = self.get_deep_linked_assignment_configuration(request)
+
+        if (
+            # We found an assignment that corresponds to this launch on the DB
+            assignment
+            # that assignment was originally deep linked
+            and assignment.deep_linking_uuid
+            # the UUID we generated on the deep link doesn't match our record in the DB
+            and assignment.deep_linking_uuid
+            != deep_linked_config.get("deep_linking_uuid")
+        ):
+            # The assignment must have been re-deep-linked in Moodle and our DB record is outdated
+            # Update our uuid so we can trust our DB again
+            assignment.deep_linking_uuid = deep_linked_config.get("deep_linking_uuid")
+            # Get the config from the DL
+            return self._assignment_config_from_deep_linked_config(deep_linked_config)
+
+        if assignment:
+            # In other cases, if we have a record of the assignment in our DB, trust that info.
+            return self._assignment_config_from_assignment(assignment)
+
+        # In other LMSes we'd look at historical_assignment here.
+        # Moodle doesn't support resource_link_id so we will never have an historical_assignment
+
+        # If we don't have an assignment in the DB this means
+        # - This is the first launch of a deep linked assignment, get the info from the DL.
+        # - This install doesn't use deep linking, rely on `.get`'s default to return None.
+        #     That will make this an "un_configured assignment" and we'll show the file picker.
+        return self._assignment_config_from_deep_linked_config(deep_linked_config)
```

Comment:
This is the same as the first `if`'s early return, right?

We might want to consider merging both conditions, but I reckon having this separated with its own comment helps to better understand the flow.